pennylane/devices/qubit/sampling.py

```diff
@@ -43,9 +43,13 @@ def measure_with_samples(
     # if the measurement process involves a Sum or a Hamiltonian, measure each
     # of the terms separately and sum
     if isinstance(mp, ExpectationMP) and isinstance(mp.obs, Hamiltonian):
+        coeffs, terms = mp.obs.terms()
+        if qml.math.get_interface(coeffs[0]) == "tensorflow":
+            vals = [measure_with_samples(ExpectationMP(t), state, shots, rng=rng) for t in terms]
+            return qml.math.sum(qml.math.multiply(coeffs, vals))
         return sum(
```

Comment:
I'm not sure if this is the best way to deal with this, but I had to use `qml.math` in the case of Tensorflow because otherwise, list comprehension/iteration fails [like this](https://stackoverflow.com/questions/62068323/iterating-over-tf-tensor-is-not-allowed-autograph-is-disabled-in-this-function). We can always do this if we prefer consistency, but I'm not sure if this will come with a performance impact

Reply:
I locally updated `Hamiltonian.terms` to return:
```
        return self.data, self.ops
```
instead of using `self.coeffs`, and then I was able to comment these lines out.

Reply:
all tests passed! 🎉